src/application/extension-popup/views/customization-settings/customization-settings.tsx

```diff
@@ -0,0 +1,94 @@
+import { FormProvider, useForm } from 'react-hook-form';
+import { useUpdateEffect } from 'react-use';
+import isEqual from 'lodash/isEqual';
+import { useMemo, useRef } from 'react';
+
+import {
+  EXTENSION_POPUP_ROUTE,
+  ExtensionSettings,
+  useExtensionPopup,
+  useExtensionSettings,
+} from 'shared/extension';
+import { IconButton } from 'shared/ui';
+
+import { settingListItemGroups } from './constants';
+import { CustomizationSettingsGroup } from './customization-settings-group';
+
+type ExtensionSettingsFormValues = Omit<
+  ExtensionSettings,
+  'entire-extension-enabled'
+>;
+
+export const CustomizationSettingsView = () => {
+  const extensionPopup = useExtensionPopup();
+  const { changeExtensionSetting, extensionSettings } = useExtensionSettings();
+
+  const isExtensionEnabled = useMemo(() => {
+    return extensionSettings['entire-extension-enabled'];
+  }, [extensionSettings]);
+
+  const customizationSettings = useMemo(() => {
+    const { 'entire-extension-enabled': _, ...customizationSettings } =
+      extensionSettings;
+
+    return customizationSettings;
+  }, [extensionSettings]);
+
+  const previousFormValues = useRef<ExtensionSettingsFormValues>(
+    customizationSettings,
+  );
```

Comment:
can you please add descriptive comment why do we need this?
I will probably forgot why we need this workaround in a couple of days :P

Reply:
good idea